web/src/settings_profile_fields.js

```diff
@@ -29,6 +29,125 @@ function display_success_status() {
+
+function set_up_select_field_edit_form_events($profile_field_form) {
+    toggle_alphabetize_btn($(".profile-field-form"));
+    $profile_field_form
+        .find(".edit_profile_field_choices_container")
+        .on("input", ".choice-row input", () => {
+            toggle_alphabetize_btn($(".profile-field-form"));
+        });
+
+    $profile_field_form
+        .find(".edit_profile_field_choices_container")
+        .on("input", ".choice-row input", (e) => {
+            add_choice_row(e);
+            toggle_alphabetize_btn($(".profile-field-form"));
+        });
+
+    $profile_field_form
+        .find(".edit_profile_field_choices_container")
+        .on("click", "button.delete-choice", (e) => {
+            delete_choice_row(e);
+            toggle_alphabetize_btn($(".profile-field-form"));
+        });
+}
+
+function toggle_alphabetize_btn($container) {
+    const $choices = $container.find(".choice-row input");
+    const choices_array = $choices
+        .map(function () {
+            return $(this).val().trim().toLowerCase();
+        })
+        .get()
+        .filter(Boolean);
+    const sorted_choices = [...choices_array].sort();
+
+    const are_choices_sorted = JSON.stringify(choices_array) === JSON.stringify(sorted_choices);
+
+    if (are_choices_sorted) {
+        $container.find("#alphabetize-choices-btn").prop("disabled", true).hide();
+    } else {
+        $container.find("#alphabetize-choices-btn").prop("disabled", false).show();
+    }
+}
+
+function sorting_field_choices_data(field_data) {
+    const sorted_data_array = Object.entries(field_data).sort((a, b) =>
+        a[1].text.toLowerCase().localeCompare(b[1].text.toLowerCase()),
+    );
+
+    const sorted_data = {};
+    let index = 1;
+    for (const [key, value] of sorted_data_array) {
+        value.order = index.toString();
+        sorted_data[key] = value;
+        index = index + 1;
+    }
+
+    return sorted_data;
+}
+
+function alphabetize_choices() {
+    const field_data = [];
+
+    // Collect input values into an array
+    $(".new-profile-field-form")
+        .find("div.choice-row input")
+        .each(function () {
+            const text = $(this).val().trim();
+            if (text) {
+                field_data.push(text);
+            }
+        });
+
+    // Sort the array
+    field_data.sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
+
+    // Update input values with sorted array
+    $(".new-profile-field-form")
+        .find("div.choice-row input")
+        .each(function (index) {
+            $(this).val(field_data[index] || ""); // Set empty string if no corresponding value in sorted array
+        });
+
+    $("#alphabetize-choices-btn").prop("disabled", true).hide();
+}
+
+function alphabetize_choices_edit_form($profile_field_form) {
+    let field_data = {};
+    field_data = read_select_field_data_from_form($profile_field_form);
+
+    const alphabetized_field_data = sorting_field_choices_data(field_data);
```

Comment:
Can't we directly sort the fields as per the data in the input like we do in the stream creation form?